aiohttp/client.py

```diff
@@ -543,7 +544,13 @@ async def _request(
 
                     if auth is None:
                         auth = auth_from_url
-                    if auth is None:
+
+                    # If the session has a _base_url set and this is a redirect to a different origin
+                    # we should not set the auth
+                    is_auth_reset_needed = (
+                        is_redirect_to_different_origin and self._base_url is not None
+                    )
+                    if auth is None and not is_auth_reset_needed:
```

Comment:
Rather than tracking this redirect variable, I'm wondering if we should just do something like `(not self._base_url or self._base_url.origin() == url.origin())`?

I was also thinking that maybe we might change the base_url behaviour so that you can still make requests to absolute URLs. If we do that, then this check would ensure that we're not sending the auth when we request a different domain to what is defined in the base_url.

Reply:
Yes, that makes sense. Fixed it.